File: ros_x/catkin_ws/src/beginner_tutorials/src/talker.cpp
/*
 * Taken from
 * http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28c%2B%2B%29
 */

#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

int main( int argc,
          char** argv )
{
    ros::init( argc,
               argv,
               "talker" );

    ros::NodeHandle n;

    ros::Publisher chatter_pub = n.advertise<std_msgs::String>( "chatter",
                                                                1000 );

    ros::Rate loop_rate( 10 );

    int message_count = 0;

    while( ros::ok() )
    {
        std_msgs::String msg;

        std::stringstream ss;
        ss << "hello ros" << message_count;
        msg.data = ss.str();

        ROS_INFO( "%s", msg.data.c_str() );

        chatter_pub.publish( msg );

        ros::spinOnce();

        loop_rate.sleep();

        ++message_count;
    }

    return 0;
}

File: ros_x/catkin_ws/src/beginner_tutorials/scripts/talker.py
#!/usr/bin/env python
# From http://wiki.ros.org/ROS/Tutorials/WritingPublisherSubscriber%28python%29
# Publishes information to a ROS topic.

import rospy
from std_msgs.msg import String

def talker():
    pub = rospy.Publisher( 'chatter',
                           String,
                           queue_size = 10 )

    rospy.init_node( 'talker',
                     anonymous = True )

    rate = rospy.Rate(10) # 10 Hz

    while not rospy.is_shutdown():
        hello_str = "hello ros %s" % rospy.get_time()
        rospy.loginfo( hello_str )
        pub.publish( hello_str )
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
